player/lua/osc.lua

```diff
@@ -2163,7 +2163,7 @@ function hide_osc()
         -- typically hide happens at render() from tick(), but now tick() is
         -- no-op and won't render again to remove the osc, so do that manually.
         state.osc_visible = false
-        render_wipe()
+        set_osd(osc_param.playresy, osc_param.playresy, "")
```

Comment:
This is an incorrect solution. `render_wipe` calls `state.osd:remove()` which goes into mpv and can perform cleanups related to bitmap memory allocations etc.

The correct solution is to reset `state.osd.data` to `nil` or an empty string at `render_wipe` so that it's both cleaned up inside mpv and also `set_osd` won't ignore any following call with some actual text. Empty string is slightly less correct but more safe, in case someone calls `state.osd:update()` without setting `.data` to some real text, so let's go with an empty string.

While at it, there's another place at the code (before your patch) which also incorrectly cleans the rendering by using an empty string. It should be changed to use `render_wipe` too. You can do both in the same commit.

Reply:
How does it look now?